qutebrowser/browser/commands.py

```diff
@@ -643,14 +643,32 @@ def scroll_perc(self, perc: {'type': float}=None,
 
     @cmdutils.register(instance='command-dispatcher', hide=True,
                        scope='window', count='count')
-    def scroll_page(self, x: {'type': float}, y: {'type': float}, count=1):
+    def scroll_page(self, x: {'type': float}, y: {'type': float}, *,
+                    top_navigate: {'type': ('prev', 'decrement')}=None,
+                    bottom_navigate: {'type': ('next', 'increment')}=None,
+                    count=1):
         """Scroll the frame page-wise.
 
         Args:
             x: How many pages to scroll to the right.
             y: How many pages to scroll down.
+            bottom_navigate: :navigate to the next page on bottom
+            top_navigate: :navigate to the previous page on top
             count: multiplier
         """
+        frame = self._current_widget().page().currentFrame()
+        if not frame.url().isValid():  # Issue 701
+            return
```

Comment:
I'm not sure if this is actually the right thing to do.

Could it be possible that the error from #701 only showed up when you're at the bottom of the page, so `self.navigate` would get called? That's the first thing I see which actually *needs* an URL.

If that really gets invoked and there is no URL set yet, the error is right and it should be shown rather than effectively being silenced here.

If there *should* be an URL and you get this error, there's some actual issue in #701 and that should be fixed (rather than silenced here :wink:).